lib/r10k/action/deploy/display.rb

```diff
@@ -52,7 +52,8 @@ def source_info(source, argv=[])
           source_info[:prefix] = source.prefix if source.prefix
           source_info[:remote] = source.remote if source.respond_to?(:remote)
 
-          env_list = source.environments.select { |env| argv.empty? || argv.include?(env.name) }
+          select_all_envs = requested_environments.empty?
+          env_list = source.environments.select { |env| select_all_envs || request_environments.include?(env.name) }
```

Comment:
```suggestion
          env_list = source.environments.select { |env| select_all_envs || requested_environments.include?(env.name) }
```

seems Not Great that none of the tests caught this.. Is there something we could add?

Reply:
_ouch_  I'll look

Reply:
I've added a test to check that it can at least meaningfully collect source and environment info.